extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/serialization/JsonbSerde.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.kafka.client.serialization;
+
+import java.util.Map;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+
+/**
+ * A {@link Serde} that (de-)serializes JSON using JSON-B.
+ */
+public class JsonbSerde<T> implements Serde<T> {
+
+    private final Class<T> type;
+    private final Jsonb jsonb;
+
+    public JsonbSerde(Class<T> type) {
+        this(type, JsonbBuilder.create());
+    }
+
+    public JsonbSerde(Class<T> type, Jsonb jsonb) {
+        this.type = type;
+        this.jsonb = jsonb;
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+    }
+
+    @Override
+    public void close() {
+        try {
+            jsonb.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Serializer<T> serializer() {
+        return new JsonbSerializer<T>();
```

Comment:
should we pass the jsonb object here?

Reply:
Agh, yes. Didn't pay enough attention when promoting it from being an inner class.